spyder/config/user.py

```diff
@@ -390,7 +390,13 @@ def get(self, section, option, default=NoDefault):
             if PY2:
                 try:
                     value = value.decode('utf-8')
-                except (UnicodeEncodeError, UnicodeDecodeError):
+
+                    # Some str config values expect to be eval after decoding
+                    new_value = ast.literal_eval(value)

```

Comment:
You should nest a new try/except instead of adding extra exceptions to the already existing outer try/except